source/common/utils/logging.js

```diff
@@ -1,6 +1,78 @@
 // @flow
+import { isEmpty } from 'lodash';
+import type {
+  FormatMessageContextParams,
+  ConstructMessageBodyParams,
+  MessageBody
+} from '../types/logging.types';
+
+const DEFAULT_MESSAGE_BODY: MessageBody = {
+  at: '',
+  env: '',
+  ns: [
+    'daedalus',
+  ],
+  data: {},
+  app: [
+    'daedalus'
+  ],
+  msg: '',
+  pid: '',
+  sev: '',
+  thread: ''
+};
+
 export const stringifyData = (data: any) => JSON.stringify(data, null, 2);
 
 export const stringifyError = (error: any) => (
   JSON.stringify(error, Object.getOwnPropertyNames(error), 2)
 );
+
+export const formatContext = (context: FormatMessageContextParams): string => {
+  const { appName, electronProcess, level, network } = context;
+  return `[${appName}.*${network}*:${level}:${electronProcess}]`;
+};
+
+export const formatMessageTime = (date: Date): string => {
+  const [year, time] = date.toISOString().split('T');
+  return `[${year} ${time.slice(0, -1)} UTC]`;
+};
+
+export const constructMessageBody = (bodyData: ConstructMessageBodyParams): MessageBody => {
+  const { data = {} } = bodyData;
+  let messageBody = { ...bodyData, data };
+
+  if (typeof data === 'object' && !isEmpty(data)) {
+    messageBody = { ...messageBody, data };
+  }
+
+  if (typeof data === 'string' && data !== ' ') {
+    messageBody = { ...messageBody, data: { response: data } };
+  }
+
+  return { ...DEFAULT_MESSAGE_BODY, ...messageBody };
+};
+
+export const formatMessage = (message: Object): string => {
+  const at = message.date.toISOString();
```

Comment:
Why does that pass flow checks if you use the generic `Object` type and then access `date` prop etc. without null checks?